src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -219,37 +288,90 @@ private static MethodHandle createRepeatIndexSwitch(MethodHandles.Lookup lookup,
+
+        MethodHandle body;
+        if (!selectorType.isPrimitive()) {
+            var castedNullCheckMT = MethodHandles.explicitCastArguments(NULL_CHECK,
+                    MethodType.methodType(boolean.class, selectorType));
+            body = MethodHandles.guardWithTest(MethodHandles.dropArguments(castedNullCheckMT, 0, int.class),
+                    MethodHandles.dropArguments(MethodHandles.constant(int.class, -1), 0, int.class, selectorType),
+                    mainTest);
+        } else {
+            body = mainTest;
+        }
+
         MethodHandle switchImpl =
-                MethodHandles.permuteArguments(body, MethodType.methodType(int.class, Object.class, int.class), 1, 0);
+                MethodHandles.permuteArguments(body, MethodType.methodType(int.class, selectorType, int.class), 1, 0);
         return withIndexCheck(switchImpl, labels.length);
     }
 
+    private static boolean floatEqCheck(float value, Float constant) {
+        if (Float.valueOf(value).equals(constant.floatValue())) {
+            return true;
+        }
+        return false;
+    }
+
+    private static boolean doubleEqCheck(double value, Double constant) {
+        if (Double.valueOf(value).equals(constant.doubleValue())) {
+            return true;
+        }
+        return false;
+    }
+
+    private static boolean booleanEqCheck(boolean value, Boolean constant) {
+        if (Boolean.valueOf(value).equals(constant.booleanValue())) {
+            return true;
+        }
+        return false;
+    }
+
     private static boolean integerEqCheck(Object value, Integer constant) {
         if (value instanceof Number input && constant.intValue() == input.intValue()) {
             return true;
         } else if (value instanceof Character input && constant.intValue() == input.charValue()) {
             return true;
         }
-
+        else if (value instanceof Boolean input && constant.intValue() == (input.booleanValue() ? 1 : 0)) {
+            return true;
+        }
         return false;
     }
 
     private static boolean isZero(int value) {
         return value == 0;
     }
 
+    private static boolean unconditionalExactnessCheck(Object selectorType, Class<?> targetType) {
+        if ((selectorType.equals(byte.class) && targetType.equals(Byte.class)) ||
+            (selectorType.equals(char.class) && targetType.equals(Character.class)) ||
+            (selectorType.equals(long.class) && targetType.equals(Long.class)) ||
+            (selectorType.equals(double.class) && targetType.equals(Double.class)) ||
+            (selectorType.equals(float.class) && targetType.equals(Float.class)) ||
+            (selectorType.equals(short.class) && targetType.equals(Short.class)) ||
+            (selectorType.equals(int.class) && targetType.equals(Integer.class)))
+            return true;
+        else if (selectorType.equals(targetType) || (selectorType.equals(byte.class) && !targetType.equals(char.class)  ||
```

Comment:
this code reminds me of some tricks we did to simplify similar code in the past, see `com.sun.tools.javac.code.TypeTag` and related:  `com.sun.tools.javac.code.TypeTag.NumericClasses` and method `com.sun.tools.javac.code.TypeTag::isSubRangeOf` and how it is used to determine if a numerical type is subtype of another numerical type. These tricks could be reused here, but up to you. I could also be done in a future refactoring.

Reply:
Thanks for the recommendation. I used them and ported the subrange functionality to Wrappers (added two extra parameters for the enum variants).